### PR TITLE
Add metrics dashboard #173119847

### PR DESCRIPTION
**Pull request by bencalegari**
- Metrics is password protected when envs have AUTH_USERNAME/PASSWORD

[#173119847]

Co-authored-by: Alex Gonzalez <agonzalez@codeforamerica.org>

### CODE REVIEW

File: app/views/metrics/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  <div class="grid__item width-one-third">
+    <div class="statistic-card">
+      <p class="statistic-card__label">Ok</p>
+      <p class="statistic-card__number"><%= @metrics.application_experience(:ok) %> 😐</p>
+    </div>
+  </div>
+  <div class="grid__item width-one-third">
+    <div class="statistic-card">
+      <p class="statistic-card__label">Bad</p>
+      <p class="statistic-card__number"><%= @metrics.application_experience(:bad) %> 😠</p>
```
Comment (luigi) [inline app/views/metrics/index.html.erb:61]: For consistency, we should use the same emoji as on the success screen, which is 🙁 .

**Comment by luigi** [comment]
Thanks for catching that AUTH_USERNAME/PASSWORD issue, @MatthewSchultz. Because this feature is meant primarily for production (which won't have sitewide auth), I think we should keep a second set of ENV vars since we wouldn't care about setting them for demo and staging. I suggest `METRICS_USERNAME` and `METRICS_PASSWORD`. Then this `MetricsController` can have its own auth check rather than using the sitewide concern.

**Comment by MatthewSchultz** [comment]
That certainly can work, yes, and the sitewide concern still functions in this case.

File: app/controllers/metrics_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  before_action :authenticate
+
+  def index
+    @metrics = MetricsReport.new
+  end
+
+  def authenticate
+    authenticate_or_request_with_http_basic do |username, password|
+      username == ENV['METRICS_USERNAME'] && password == ENV['METRICS_PASSWORD']
```
Comment (luigi) [inline app/controllers/metrics_controller.rb:14]: If these ENV vars aren't set, would the auth modal pop up but it'd be looking for blank strings? Maybe first have `return if ENV['METRICS_USERNAME'].blank?` like in the concern.

Comment (bencalegari) [reply app/controllers/metrics_controller.rb:14]: I thought so but apparently not. I realized the test I wrote still assumed the vars were set. But `return`ing when it's blank just lets it continue without authorization. I'll need to look at this further.

Comment (MatthewSchultz) [reply app/controllers/metrics_controller.rb:14]: I THINK it works without the return. username/pw when not set by the header are nil, and ENV[stuff], when not set, is nil, so we get `nil == nil && nil == nil` - which...is true.

I put the return in there mostly to clarify what I was doing to begin with (I find it easier to read and to follow).

**Pull request merged**